### PR TITLE
Fix: Added error handling and retry logic in on_message_edit event

### PR DESCRIPTION
**Pull request by codewithriza**
This PR adds error handling and retry logic to the `on_message_edit` event in the Cog class. If a `TimeoutError` occurs during database operations, the function waits for 1 second and retries the operation. This helps prevent issues with the database connection being closed unexpectedly.

### Changes
- Added error handling and retry logic to `on_message_edit`
- Added `try-except` block to catch `TimeoutError` and other exceptions
- Added `await asyncio.sleep(1)` for retrying the operation

Solves issue [#0101](https://github.com/DTS-11/PizzaHat/issues/46)

### CODE REVIEW

File: PizzaHat/cogs_hidden/guild_logs.py [hunk 1]
```diff
@@ -40,11 +42,12 @@ async def check_log_enabled(
 
     # ====== MESSAGE LOGS ======
 
-    @Cog.listener()
-    async def on_message_edit(self, before: discord.Message, after: discord.Message):
-        if not before.guild or not after.guild:
-            return
+@Cog.listener()
```
Comment (fuh-Q) [inline PizzaHat/cogs_hidden/guild_logs.py:45]: by removing the indentation here, you're completely detaching the listener from the cog

File: PizzaHat/cogs_hidden/guild_logs.py [hunk 2]
```diff
@@ -73,6 +76,16 @@ async def on_message_edit(self, before: discord.Message, after: discord.Message)
...
+        # Retry logic: wait for a short period and then retry
+        await asyncio.sleep(1)
+        await self.on_message_edit(before, after)  # Retry the operation
+    
+    except Exception as e:
+        # Handle other exceptions
+        print(f"An error occurred: {e}")
+
 
     @Cog.listener()
```
Comment (fuh-Q) [inline PizzaHat/cogs_hidden/guild_logs.py:90]: ...and all commands, functions and methods that are defined going forward will be local to this listener's callback function, and will not be associated with the cog either

**Review by fuh-Q — COMMENTED** [review]
indentation changes breaks everything

**Review by fuh-Q — COMMENTED** [review]
indentation changes break everything

File: PizzaHat/cogs_hidden/guild_logs.py (hunk 2 above)
Comment (codewithriza) [reply PizzaHat/cogs_hidden/guild_logs.py:90]: alr thanks

**Comment by fuh-Q** [comment]
just indent the listener one level in, and that'll fix it

**Comment by DTS-11** [comment]
this issue has been solved

**Pull request closed without merging**